app/serializers/site_serializer.rb

```diff
@@ -250,9 +253,37 @@ def displayed_about_plugin_stat_groups
     About.displayed_plugin_stat_groups
   end
 
+  SIDEBAR_TOP_TAGS_TO_SHOW = 5
+
+  def navigation_menu_site_top_tags
+    if top_tags.present?
+      tag_names = top_tags[0...SIDEBAR_TOP_TAGS_TO_SHOW]
+      serialized = serialize_tags(Tag.where(name: tag_names))
```

Comment:
This is an extra query on page load but it is limited to 5 objects so it wouldn't add much overhead.